### DISCUSSION TITLE
output highlight result as LaTeX like pygmentize?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [tree-sitter/tree-sitter#3390] [Ideas] [answered]
```sh
❯ pygmentize -L formatters
Pygments version 2.17.2, (c) 2006-2023 by Georg Brandl, Matthäus Chajdas and contributors.

Formatters:
~~~~~~~~~~~
* html:
    Format tokens as HTML 4 ``<span>`` tags. By default, the content is enclosed in a ``<pre>`` tag, itself wrapped in a ``<div>`` tag (but see the `nowrap` option). The ``<div>``'s CSS class can be set by the `cssclass` option. (filenames *.html, *.htm)

...

* latex, tex:
    Format tokens as LaTeX code. This needs the `fancyvrb` and `color` standard packages. (filenames *.tex)

...

* terminal, console:
    Format tokens with ANSI color sequences, for output in a text console. Color sequences are terminated at newlines, so that paging the output works correctly.

...

* text, null:
    Output the text unchanged without any formatting. (filenames *.txt)
```

Looks like `tree-sitter highlight` only support `terminal`.

How about let tree-sitter output highlight result as LaTeX?

**Comment by amaanq** [comment]
`tree-sitter highlight -H`

**Reply by Freed-Wu** [reply]
Can we have tree-sitter hightligh -L for LaTeX?

**Reply by amaanq** [reply, marked as answer]
I'm not implementing that - pr welcome.

**Reply by Freed-Wu** [reply]
I take a look at HTMLRenderer. If we want to realize a new TeXRenderer. Do you mind we create a new trait Renderer and then let HTMLRenderer, TeXRenderer, ... to impl it? Because HTMLRenderer and TeXRenderer have some similar methods, such as HTML need escape some character, TeX alsoe need escape some character, too. The difference is HTML escape is deterministic: `<` -> `&lt;`, TeX escape can be controled: `<` -> `\TSZlt{}`. the `TSZ` can be change by `\def\TSZlt{\char`\<}`

@amaanq 

Related PR: #5846
